src/devices/Max7219/MatrixTextWriter.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading;
+
+namespace Iot.Device.Max7219
+{
+
+    /// <summary>
+    /// Matrix text writer.
+    /// </summary>
+    public class MatrixTextWriter
+    {
+        readonly Max7219 _device;
+
+        public MatrixTextWriter(Max7219 device, Font font)
+        {
+            _device = device;
+            Font = font;
+        }
+
+        public Font Font { get; set; }
+
+        /// <summary>
+        /// Writes a char to the given device with the specified font.
+        /// </summary>
+        public void WriteLetter(int deviceId, char chr, bool flush = true)
+        {
+            var bitmap = Font.GetItem(chr);
+            var end = Math.Min(bitmap.Length, Max7219.NumDigits);
+            for (int col = 0; col < end; col++)
+            {
+                _device[deviceId, col] = bitmap[col];
+            }
+            if (flush)
+            {
+                _device.Flush();
+            }
+        }
+
+        /// <summary>
+        ///  Scrolls the underlying buffer (for all cascaded devices) up one pixel
+        /// </summary>
+        public void ScrollUp(bool flush)
```

Comment:
>flush [](start = 34, length = 5)

why flush parameter not defaulted to true as we did in WriteLetter?

Reply:
this comment apply to the rest of the methods takes flush

---
In reply to: [245435432](https://github.com/dotnet/iot/pull/104#discussion_r245435432) [](ancestors = 245435432)